vault/seal_autoseal.go

```diff
@@ -147,6 +147,69 @@ func (d *autoSeal) GetStoredKeys(ctx context.Context) ([][]byte, error) {
 	return keys, nil
 }
 
+func (d *autoSeal) upgradeStoredKeys(ctx context.Context) error {
+	pe, err := d.core.physical.Get(ctx, StoredBarrierKeysPath)
+	if err != nil {
+		return errwrap.Wrapf("failed to fetch stored keys: {{err}}", err)
+	}
+
+	// This is not strictly an error; we may not have any stored keys, for
+	// instance, if we're not initialized
+	if pe == nil {
+		return nil
+	}
+
+	blobInfo := &physical.EncryptedBlobInfo{}
+	if err := proto.Unmarshal(pe.Value, blobInfo); err != nil {
+		return errwrap.Wrapf("failed to proto decode stored keys: {{err}}", err)
+	}
+
+	if blobInfo.KeyInfo != nil && blobInfo.KeyInfo.KeyID != d.Access.KeyID() {
+		d.core.logger.Info("autoseal: upgrading stored keys")
```

Comment:
This log will be formatted incorrectly since autoseal isn't part of the name. We may instead want to hand autoSeal its own logger that can be named specifically for this use.

Reply:
or just do a one off `d.core.logger.Named("autoseal").Info(...)`

Reply:
Will do